### PR TITLE
BUGFIX: RAIL-1869 Fix Date Filter styling issues and cross-compability issues, RAIL-2233 Fix Visualization rendering after login failed

### PR DESCRIPTION
**Pull request by Tulean**
---

[Check PR owner responsibilities](https://confluence.intgdc.com/display/Development/Code-reviews#Code-reviews-Ownerresponsibilities)

Supported PR commands:

Command | Description
--- | ---
`extended test - examples` | Live examples tests
`extended test - storybook` | Storybook screenshot tests

See more [options](https://confluence.intgdc.com/display/kbhr/How+to+work+with+git+and+Github#HowtoworkwithgitandGithub-Extendedchecks).

# PR Checklist

- [ ] Verify code changes ([Checklist](https://confluence.intgdc.com/display/Development/Code-reviews+checklist), [Best practices](https://confluence.intgdc.com/display/Development/Code-reviews+best+practices))
- [ ] [Verify pull-request formalities](https://confluence.intgdc.com/display/Development/Code-reviews)
- [x] Change was tested by using [gdc-dev-release](https://confluence.intgdc.com/display/~tomas.vojtasek/Private+NPM) in [gdc-analytical-designer](https://github.com/gooddata/gdc-analytical-designer) and [gdc-dashboards](https://github.com/gooddata/gdc-dashboards) (if applicable)
- [ ] Migration guide (for major changes) is mentioned in [CHANGELOG.md](../blob/master/CHANGELOG.md).
- [x] Successful `extended test - examples`
- [x] Successful `extended test - storybook`
- [ ] Checked yarn.lock consistency (no dep. duplicities especially Goodstrap)

# Related PRs

- gdc-analytical-designer: https://github.com/gooddata/gdc-analytical-designer/pull/3033
- gdc-dashboards: https://github.com/gooddata/gdc-dashboards/pull/2749

# Related Jira tasks

### CODE REVIEW

**Comment by Tulean** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7064/ : SUCCESS in 6m 41s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3009/ : SUCCESS in 4s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3694/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3010/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3063/ : SUCCESS in 18m 07s

**Comment by Tulean** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1549.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3011/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3064/ : FAILURE in 19m 21s

**Comment by Tulean** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1549.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3012/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3065/ : SUCCESS in 6m 14s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7075/ : SUCCESS in 3m 15s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3023/ : SUCCESS in 3s

**Comment by Tulean** [comment]
extended test - storybook

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7079/ : SUCCESS in 5m 08s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3030/ : SUCCESS in 7s

**Comment by yenkins** [comment]
Link to Storybook test result:

https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3704/artifact/test-storybook/html-report/index.html

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3031/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3074/ : SUCCESS in 24m 45s

**Comment by Tulean** [comment]
extended test

**Comment by yenkins** [comment]
Demo is running at https://client-demo.na.intgdc.com:50039/storybook/

**Comment by gdgate** [comment]
Build failed (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3036/ : SUCCESS in 10s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3077/ : FAILURE in 12m 16s

**Comment by Tulean** [comment]
extended test - examples

**Comment by yenkins** [comment]
:arrow_right: Started Live Examples deploy to:
https://gdc-live-examples-1549.herokuapp.com

It will be destroyed on merge/close. See [ci-infra](https://github.com/gooddata/ci-infra/blob/master/jenkins/jobs/templates/orchestrators/client-pr-dispatcher.groovy) and [gdc-ci.yaml](https://github.com/gooddata/gooddata-react-components/blob/master/gdc-ci.yaml).

**Comment by gdgate** [comment]
Build succeeded (check-extended pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3037/ : SUCCESS in 8s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-pull-request-dispatcher/3078/ : SUCCESS in 28m 21s

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7088/ : SUCCESS in 7m 07s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-oss-check/3051/ : SUCCESS in 3s

File: src/components/filters/DateFilter/ExcludeCurrentPeriodToggle/ExcludeCurrentPeriodToggle.tsx [hunk 1]
```diff
@@ -25,7 +25,10 @@ export const ExcludeCurrentPeriodToggle: React.FC<IExcludeCurrentPeriodTogglePro
 }) => (
     <div className="gd-extended-date-filter-exclude-current">
         <BubbleHoverTrigger>
-            <label className={cx("s-exclude-current-period", "input-checkbox-label")}>
+            <label
+                className={cx("s-exclude-current-period", "input-checkbox-label")}
+                style={disabled ? { color: "#b0beca" } : { color: "#464e56" }}
```
Comment (no23reason) [inline src/components/filters/DateFilter/ExcludeCurrentPeriodToggle/ExcludeCurrentPeriodToggle.tsx:30]: Can we create a CSS class for this?

File: src/components/filters/DateFilter/DateFilterBody/DateFilterBody.tsx [hunk 2]
```diff
@@ -105,66 +105,68 @@ export class DateFilterBody extends React.Component<IDateFilterBodyProps, IDateF
...
-                        // tslint:disable-next-line:jsx-no-lambda
-                        onClick={() => {
-                            onCancelClick();
-                            closeDropdown();
-                        }}
-                    >
-                        {dateFilterButton}
-                    </div>
-                )}
+            <div className="gd-extended-date-filter-container" style={{ display: "flex", flex: "1 1 auto" }}>
```
Comment (no23reason) [inline src/components/filters/DateFilter/DateFilterBody/DateFilterBody.tsx:108]: Can we create a CSS class for this?

**Comment by gdgate** [comment]
Build in pipeline check aborted.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-storybook-zuul-docker/3707/ : SUCCESS in 18m 00s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-unit-tests-zuul-docker-el7/7090/ : SUCCESS in 5m 25s
- https://checklist.intgdc.com/job/compliance/job/gooddata-react-components-merge-compliance-check/1716/ : SUCCESS in 19s
- https://checklist.intgdc.com/job/client-libs/job/gooddata-react-components-examples-zuul-docker/2513/ : SUCCESS in 21m 33s

**Pull request merged**